hello/hello_continue_as_new.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.worker import Worker
+
+
+@dataclass
+class ComposeGreetingInput:
+    greeting: str
+    name: str
+
+
+@activity.defn
+async def compose_greeting(input: ComposeGreetingInput) -> str:
+    return f"{input.greeting}, {input.name}!"
+
+
+@workflow.defn
+class GreetingWorkflow:
+    def __init__(self) -> None:
+        self.exit_requested = False
+
+    @workflow.run
+    async def run(self, name: str) -> str:
+        # Invoke activity (or do other meaningful work)
+        result = await workflow.execute_activity(
+            compose_greeting,
+            ComposeGreetingInput("Hello", name),
+            start_to_close_timeout=timedelta(seconds=10),
+        )
+
+        # Continue as new every 2 seconds until we are told to complete. This is
+        # just a demonstration. In production code you might want to continue as
+        # new when the history is growing too large. Continue as new is
+        # essentially a workflow restart (though another workflow can be used).
```

Comment:
I don't think this is a good sample for continue as new, we should show actual code that needs it instead

Reply:
I like the TS one better, it's much simpler and helps demonstrate that CAN is basically recursion.

Another good sample could be https://docs.temporal.io/typescript/workflows#example, I've recommended this pattern numerous times in the past.

Reply:
Updated to be the same as the TS one